src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -771,8 +771,10 @@ void RaftReplDev::handle_commit(repl_req_ptr_t rreq) {
         m_listener->on_commit(rreq->lsn(), rreq->header(), rreq->key(), rreq->local_blkid(), rreq);
     }
 
-    auto prev_lsn = m_commit_upto_lsn.exchange(rreq->lsn());
-    RD_DBG_ASSERT_GT(rreq->lsn(), prev_lsn, "Out of order commit of lsns, it is not expected in RaftReplDev");
+    if (!recovery) {
```

Comment:
I didnt get this part.  
1. during recovery,  the `last_commit_index()` returns `m_commit_upto_lsn` so log reply will start from the persisted m_commit_upto_lsn, why we will trigger this assert?

2. during recovery , why we dont want to forward the `m_commit_upto_lsn` as we done replying each log?

Reply:
m_commit_upto_lsn is recovered directly from super block, it is not changed as we replay logs

Reply:
I got it after your explanation , we replay from cp_lsn which is always less than ocmmit_lsn.


Better to add a few line of comment.